jetty-core/jetty-xml/src/main/java/org/eclipse/jetty/xml/XmlConfiguration.java

```diff
@@ -465,14 +466,14 @@ public Object configure() throws Exception
                 }
                 catch (NoSuchMethodException x)
                 {
-                    throw new IllegalStateException(String.format("No matching constructor %s in %s", oClass, _configuration));
+                    throw new XmlConfigurationException("No matching constructor " + oClass);
```

Comment:
I would throw `IllegalStateException` here as before, because A) in many other places you did not convert to throw `XmlConfigurationException` (you converted just a couple), and it's caught and rethrown as `XmlConfigurationException` anyway.

Reply:
Done.